services/blockchain-indexer/shared/constants.js

```diff
@@ -84,6 +87,33 @@ const getSystemMetadata = async () => {
 	return systemMetadata;
 };
 
+const getRegisteredEndpoints = async () => {
+	if (!registeredEndpoints) {
+		registeredEndpoints = await requestConnector('getRegisteredEndpoints');
+	}
+	return registeredEndpoints;
+};
+
+const getEngineEndpoints = async () => {
+	if (!engineEndpoints) {
+		engineEndpoints = await requestConnector('getEngineEndpoints');
+	}
+	return engineEndpoints;
+};
+
+const getAllRegisteredEndpoints = async () => {
+	if (!allRegisteredEndpoints) {
+		registeredEndpoints = await getRegisteredEndpoints();
+		engineEndpoints = await getEngineEndpoints();
+
+		allRegisteredEndpoints = engineEndpoints
+			.map(e => e.name)
+			.concat(registeredEndpoints);
+	}
+
+	return allRegisteredEndpoints;
```

Comment:
will it be better to make it a hash table (i.e object or Set) instead so that we can do constant loopup? same for the cached registered endpoints and engine endpoints.